### PR TITLE
Docs notes

### PR DESCRIPTION
**Pull request by mprimi**
This is **NOT** an actual PR. I'll close it right away.

It's just a way to share some notes on the nats.go package docs (https://pkg.go.dev/github.com/nats-io/nats.go)

### CODE REVIEW

**Comment by piotrpio** [comment]
Thanks for these! I'll work on improving the docs, some really good suggestions here.

**Comment by mprimi** [comment]
@piotrpio Let me know if comments don't make sense, I went very fast through the whole package doc, so it's pretty hastily written and meant for discussion

**Comment by Jarema** [comment]
@mprimi I think we need to improve the docs a lot, and your notes can help a lot.

We should have an iterative approach to those improvements - let's make a solid run through Core NATS, but first, let's agree on the format of the docs (maybe a sample doc on one or two functions).

Also - do we want to have an example for each public method?
I find those really helping users.

**Comment by piotrpio** [comment]
@Jarema we actually already have examples for _most_ of the methods in the docs already (https://github.com/nats-io/nats.go/blob/main/example_test.go), but yes, I believe we should have an example for each public method, so I'll figure out which are not covered.
I'll create a PR with sample doc format for functions/methods/types etc. and we can continue from there.

File: nats.go [hunk 1]
```diff
@@ -609,16 +676,24 @@ type Subscription struct {
...
 //	// Terminate a message so that it is not redelivered further.
+// To just this consumer, or terminates it for all consumers of the stream?
 //	msg.Term()
 //
 //	// Signal the server that the message is being worked on and reset redelivery timer.
 //	msg.InProgress()
+
+//This section on ACKS is useful, but kinda hard to find it under Msg.
+// I have been looking at the docs daily for over a year and never know it was here.
+// Complained multiple times in these notes there's no doc for ACK behavior. I guess some exists after all.
```
Comment (piotrpio) [inline nats.go:696]: I guess ack documentation should be on `Ack()`, `Nak()` etc. methods in js.go (and there is, from what I see, although probably can be improved. `Msg` doc could simply link to those methods.

**Pull request closed without merging**